packages/zosfiles/src/api/methods/copy/Copy.ts

```diff
@@ -0,0 +1,84 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { AbstractSession, ImperativeExpect, Logger, Headers } from "@zowe/imperative";
+import { posix } from "path";
+
+import { ZosmfRestClient } from "../../../../../rest";
+import { ZosFilesConstants } from "../../constants/ZosFiles.constants";
+import { ZosFilesMessages } from "../../constants/ZosFiles.messages";
+import { IZosFilesResponse } from "../../doc/IZosFilesResponse";
+import { IHeaderContent } from "../../../../../rest/src/doc/IHeaderContent";
+/**
+ * This class holds helper functions that are used to copy the contents of datasets through the
+ * z/OSMF APIs.
+ */
+export class Copy {
+    /**
+     * Copy the contents of a dataset
+     *
+     * @param {AbstractSession}       session        - z/OSMF connection info
+     * @param {{string, string}}      fromDataSet    - The name of the data set to copy from and optionally the member name
+     * @param {{string, string}}      toDataSet      - The name of the data set to copy to and optionally the member name
+     *
+     * @returns {Promise<IZosFilesResponse>} A response indicating the status of the copying
+     *
+     * @throws {ImperativeError} Data set name must be specified as a non-empty string
+     * @throws {Error} When the {@link ZosmfRestClient} throws an error
+     *
+     * @see https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.1.0/com.ibm.zos.v2r1.izua700/IZUHPINFO_API_PutDataSetMemberUtilities.htm
+     */
+    public static async dataSet(
+        session: AbstractSession,
+        { dataSetName: fromDataSetName, memberName: fromMemberName }: { dataSetName: string, memberName?: string },
```

Comment:
Please add a separate interface for these types. Good for when someone is building the parameters outside of the call and allows for complete documentation of the properties.

Reply:
Thanks @tucker01, I have added an interface ICopyDataSet